katversion/version.py

```diff
@@ -179,13 +183,30 @@ def get_version_from_module(module):
             pass
 
 
+def _must_decode(value):
+    """Copied from pkginfo 1.4.1, _compat module."""
+    if type(value) is bytes:
+        try:
+            return value.decode('utf-8')
+        except UnicodeDecodeError:
+            return value.decode('latin1')
+    return value
+
+
 def get_version_from_unpacked_sdist(path):
     """Assume path points to an unpacked source distribution and get version."""
+    # This is a condensed version of the relevant code in pkginfo 1.4.1
     try:
-        return UnpackedSDist(path).version
-    except ValueError:
+        with open(os.path.join(path, 'PKG-INFO')) as f:
+            data = f.read()
+    except Exception as e:
```

Comment:
Not critical. 
But 'e' is unused, thus you can remove " as e".

Reply:
I copied those lines from pkginfo. Mmm, would it be safe to replace it with the more palatable `IOError` instead of the blanket `Exception`?

Reply:
That is a good suggestion, it would be much better yes.